android/src/main/java/com/oblador/keychain/cipherStorage/CipherStorageKeystoreRSAECB.java

```diff
@@ -0,0 +1,342 @@
+package com.oblador.keychain.cipherStorage;
+
+import android.Manifest;
+import android.annotation.TargetApi;
+import android.app.Activity;
+import android.app.KeyguardManager;
+import android.content.Context;
+import android.content.pm.PackageManager;
+import android.hardware.fingerprint.FingerprintManager;
+import android.os.Build;
+import android.os.CancellationSignal;
+import android.security.keystore.KeyGenParameterSpec;
+import android.security.keystore.KeyPermanentlyInvalidatedException;
+import android.security.keystore.KeyProperties;
+import android.security.keystore.UserNotAuthenticatedException;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import com.facebook.react.bridge.ReactApplicationContext;
+
+import com.oblador.keychain.exceptions.CryptoFailedException;
+import com.oblador.keychain.exceptions.KeyStoreAccessException;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.security.InvalidAlgorithmParameterException;
+import java.security.Key;
+import java.security.KeyFactory;
+import java.security.KeyPairGenerator;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.NoSuchProviderException;
+import java.security.PublicKey;
+import java.security.UnrecoverableKeyException;
+import java.security.cert.CertificateException;
+import java.security.spec.AlgorithmParameterSpec;
+import java.security.spec.KeySpec;
+import java.security.spec.X509EncodedKeySpec;
+
+import javax.crypto.Cipher;
+import javax.crypto.CipherInputStream;
+import javax.crypto.CipherOutputStream;
+import javax.crypto.KeyGenerator;
+import javax.crypto.spec.IvParameterSpec;
+
+import android.hardware.biometrics.BiometricPrompt.AuthenticationCallback;
+import moe.feng.support.biometricprompt.BiometricPromptCompat;
+import moe.feng.support.biometricprompt.BiometricPromptCompat.IAuthenticationCallback;
+
+
+@TargetApi(Build.VERSION_CODES.M)
+public class CipherStorageKeystoreRSAECB implements CipherStorage, BiometricPromptCompat.IAuthenticationCallback {
+    public static final String CIPHER_STORAGE_NAME = "KeystoreRSAECB";
+    public static final String DEFAULT_SERVICE = "RN_KEYCHAIN_DEFAULT_ALIAS";
+    public static final String KEYSTORE_TYPE = "AndroidKeyStore";
+    public static final String ENCRYPTION_ALGORITHM = KeyProperties.KEY_ALGORITHM_RSA;
+    public static final String ENCRYPTION_BLOCK_MODE = KeyProperties.BLOCK_MODE_ECB;
+    public static final String ENCRYPTION_PADDING = KeyProperties.ENCRYPTION_PADDING_RSA_PKCS1;
```

Comment:
Eventually, these really should be options.   OAEP is the default on many systems now, and is considered more secure, for example.